packages/eslint-plugin/src/rules/no-base-to-string.ts

```diff
@@ -227,7 +243,8 @@ export default createRule<Options, MessageIds>({
       const declaration = declarations[0];
       const isBaseToString =
         ts.isInterfaceDeclaration(declaration.parent) &&
-        declaration.parent.name.text === 'Object';
+        (declaration.parent.name.text === 'Object' ||
+          declaration.parent.name.text === 'Array');
```

Comment:
What is the intent behind this line? No tests change if I remove it

Reply:
This was necessary a while back (changed while merging your changes in #10432 from the `main` branch) when this check was executing sooner, potentially filtering out array types. It's a lot better now that the function checks for array/tuple types and only has this check at the very end.

I've updated the PR to match this, thanks!